src/cleanup_rules/swift/rules.toml

```diff
@@ -890,6 +890,153 @@ not_contains = [
+    ) @texp
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "texp"
+replace = "@tvalue"
+is_seed_rule = false
+#
+# Before 
+#   true == true
+#   (true) == true
+#   true == (true)
+#   (true) == (true)
+#   false == false
+#   (false) == false
+#   false == (false)
+#   (false) == (false)
+# After 
+#   true
+#
+[[rules]]
+name = "boolean_equality_expression_same_literals"
+query = """(
+    (equality_expression
+            lhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @left
+            rhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @right
+    ) @cond
+    (#eq? @left @right)
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "cond"
+replace = "true"
+is_seed_rule = false
+
+#
+# Before 
+#   true == false
+#   false == true
+#   (true) == false
+#   (false) == true
+#   true == (false)
+#   false == (true)
+#   (false) == (true)
+#   (true) == (false)
+# After 
+#   false
+#
+[[rules]]
+name = "boolean_equality_expression_opposite_literals"
+query = """(
+    (equality_expression
+            lhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @left
+            rhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @right
+    ) @cond
+    (#not-eq? @left @right)
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "cond"
+replace = "false"
+is_seed_rule = false
+
+#
+# Before 
+#   true != true
+#   (true) != (true)
+#   true != (true)
+#   (true) != true
+#   false != false
+#   (false) != false
+#   false != (false)
+#   (false) != (false)
+# After 
+#   false
+#
+[[rules]]
+name = "boolean_inequality_expression_same_literals"
+query = """(
+    (infix_expression
+            lhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @left
+            op: (custom_operator) @ops
+            rhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @right
+    ) @cond
+    (#eq? @ops "!=")
+    (#eq? @left @right)
+)"""
+groups = ["boolean_expression_simplify"]
+replace_node = "cond"
+replace = "false"
+is_seed_rule = false
+
+#
+# Before 
+#   true != true
+#   (true) != (true)
+#   true != (true)
+#   (true) != true
+#   false != false
+#   (false) != false
+#   false != (false)
+#   (false) != (false)
+# After 
+#   false
+#
+[[rules]]
+name = "boolean_inequality_expression_opposite_literals"
+query = """(
+    (infix_expression
+            lhs: [(boolean_literal) 
+                  (integer_literal)
+                  (line_string_literal)
+                 ] @left
+            op: (custom_operator) @ops
+            rhs: [(boolean_literal) 
+                  (integer_literal)
```

Comment:
Swift has 9 kinds of literals - https://github.com/alex-pinkus/tree-sitter-swift/blob/main/grammar.js#L272